baremetal/metal3machine_manager.go

```diff
@@ -418,9 +412,9 @@ func (m *MachineManager) Associate(ctx context.Context) error {
 // for the BareMetalHost through Metal3Machine. The UserDataSecretName might already be in a secret with
 // CABPK v0.3.0+, but if it is in a different namespace than the BareMetalHost,
 // then we need to create the secret.
-func (m *MachineManager) getUserDataSecretName(_ context.Context) error {
+func (m *MachineManager) getUserDataSecretName(_ context.Context) {
```

Comment:
We have two functions like this here now.

There is a choice to be made:
1. Remove the return value as PR now suggests.
2. Is there error handling missing, and the fix is to add an error condition so return value is not always nil
3. 4. Leave as is and disable linter

@kashifest @lentzi90 opinions?

Reply:
This (1) looks correct to me.
(The function did not do at all what I expected based on the name and description though :see_no_evil: )

Reply:
Yeah, it was surprising one. Some tech debt to pay here.